printer/printast_with_mappings.ml

```diff
@@ -252,10 +264,11 @@ and pattern i ppf x =
   | Ppat_lazy p ->
       line i ppf "Ppat_lazy\n";
       pattern i ppf p;
-  | Ppat_constraint (p, ct) ->
+  | Ppat_constraint (p, ct, m) ->
       line i ppf "Ppat_constraint\n";
       pattern i ppf p;
-      core_type i ppf ct;
+      option i core_type ppf ct;
```

Comment:
The `option` will print some extra `Some` and `None`, which doesn't add information.